### PR TITLE
broot: disable (impure) br shell function nag, install bash/zsh/fish files for the br function

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

I think it makes a lot of sense to disable the nag screen. I am not so sure whether we should also provide shell functions or let home-manager et al. do the work. At any rate:
    
- In bash, the user needs to source `br.sh` somewhere.
- In zsh, the user needs to `autoload -Uz br` somewhere.
- In fish, `br` should just work.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @rycee since home-manager has a broot module.

### CODE REVIEW

**Comment by lilyball** [comment]
Thank you, I've been meaning to do something similar since https://github.com/Canop/broot/issues/265 got resolved with the `BR_INSTALL` env var.

My open question was how to deal with bash/zsh. I wasn't sure if we should wrap `broot` for fish specifically to disable nagging even if the user runs `broot` directly but allow it to nag for bash/zsh, or what. It looks like you're just deferring to "the user has to handle zsh/bash themselves" which seems like probably the most reasonable solution in the absence of any direct shell support for this sort of thing.

File: pkgs/tools/misc/broot/default.nix [hunk 1]
```diff
@@ -27,6 +35,23 @@ rustPlatform.buildRustPackage rec {
...
 
   postInstall = ''
+    # Do not nag users about installing shell integration, since
+    # it is impure.
+    wrapProgram $out/bin/broot \
+      --set BR_INSTALL no
+
+    # Install shell function for bash.
+    $out/bin/broot --print-shell-function bash > br.bash
+    install -Dm0555 br.bash $out/etc/profile.d/br.sh
```
Comment (lilyball) [inline pkgs/tools/misc/broot/default.nix:45]: This needs to be sourced, not executed.

Also it looks like the function implementation uses bashisms, so we should retain the `.bash` suffix so as to not make anyone think it's POSIX-compliant sh (it uses the `function` keyword and `$(<file)`).

```suggestion
    install -Dm0444 -t $out/etc/profile.d br.bash
```

Comment (danieldk) [reply pkgs/tools/misc/broot/default.nix]: Sounds good, I have added this change.

**Review by lilyball — APPROVED** [review]
LGTM

https://github.com/NixOS/nixpkgs/pull/95209
1 package built:
broot

**Pull request merged**